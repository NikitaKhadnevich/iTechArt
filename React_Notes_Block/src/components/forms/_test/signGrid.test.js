/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import configureMockStore from 'redux-mock-store';
import SignUpGrid from '../SignInputs/SignUpGrid';

configure({ adapter: new Adapter() }); // ES6 modules

describe('SignIn', () => {
  const signProps = {
    values: {
      email: 'email',
      firstName: 'firstName',
      lastName: 'lastName',
      birthday: 'birthday',
      password: 'password',
      confirmPassword: 'confirmPassword',
    },
    errors: false,
    touched: false,
    handleChange: jest.fn(),
    isValid: false,
    setFieldTouched: jest.fn(),
    handleBlur: jest.fn(),
    handleSubmit: jest.fn(),
  };

  it('Is it SignUp Form Exist', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper).toMatchSnapshot();
    expect(signGridWrapper).toHaveLength(1);
  });

  it('Is Button Disabled', () => {
    signProps.isValid = false;
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(
      signGridWrapper.find('WithStyles(ForwardRef(Button))').props().disabled
    ).toBeTruthy();
  });
  it('Is Button Enabled', () => {
    signProps.isValid = true;
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(
      signGridWrapper.find('WithStyles(ForwardRef(Button))').props().disabled
    ).toBeFalsy();
  });

  it('Form Has email folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper.find('SignEmail').props().email).toEqual('email');
  });
  it('Form Has firstName folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper.find('SignUpFirstName').props().firstName).toEqual(
      'firstName'
    );
  });
  it('Form Has lastName folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper.find('SignUpLastName').props().lastName).toEqual(
      'lastName'
    );
  });
  it('Form Has birthday folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper.find('SignUpBirth').props().birthday).toEqual(
      'birthday'
    );
  });
  it('Form Has password folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(signGridWrapper.find('SignPassword').props().password).toEqual(
      'password'
    );
  });
  it('Form Has confirmPassword folder', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    expect(
      signGridWrapper.find('SignUpConfirmPass').props().confirmPassword
    ).toEqual('confirmPassword');
  });
  it('Form Has Correct User-data in any folders', () => {
    const signGridWrapper = shallow(<SignUpGrid {...signProps} />);
    signProps.isValid = true;
    expect(signGridWrapper.find('SignEmail').props().errors).toBeFalsy();
    expect(signGridWrapper.find('SignUpFirstName').props().errors).toBeFalsy();
    expect(signGridWrapper.find('SignUpLastName').props().errors).toBeFalsy();
    expect(signGridWrapper.find('SignUpBirth').props().errors).toBeFalsy();
    expect(signGridWrapper.find('SignPassword').props().errors).toBeFalsy();
    expect(
      signGridWrapper.find('SignUpConfirmPass').props().errors
    ).toBeFalsy();
    expect(
      signGridWrapper.find('WithStyles(ForwardRef(Button))').props().disabled
    ).toBeFalsy();
  });
});
