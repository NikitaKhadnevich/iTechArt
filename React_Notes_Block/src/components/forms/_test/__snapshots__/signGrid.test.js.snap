// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Is it SignUp Form Exist 1`] = `
<Styled(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <SignEmail
      email="email"
      errors={false}
      handleBlur={[MockFunction]}
      onChangeAccum="onChangeAccum"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    sm={6}
    xs={12}
  >
    <SignUpFirstName
      errors={false}
      firstName="firstName"
      handleBlur={[MockFunction]}
      onChangeAccum="onChangeAccum"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    sm={6}
    xs={12}
  >
    <SignUpLastName
      errors={false}
      handleBlur={[MockFunction]}
      lastName="lastName"
      onChangeAccum="onChangeAccum"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <SignUpBirth
      birthday="birthday"
      errors={false}
      handleBlur={[MockFunction]}
      onChangeAccum="onChangeAccum"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <SignPassword
      errors={false}
      handleBlur={[MockFunction]}
      onChangeAccum="onChangeAccum"
      password="password"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <SignUpConfirmPass
      confirmPassword="confirmPassword"
      errors={false}
      handleBlur={[MockFunction]}
      onChangeAccum="onChangeAccum"
      touched={false}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      color="secondary"
      disabled={true}
      fullWidth={true}
      type="submit"
      variant="outlined"
    >
      Sign Up
    </WithStyles(ForwardRef(Button))>
  </ForwardRef(Grid)>
</Styled(ForwardRef(Grid))>
`;
